libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1576,11 +1598,9 @@ void IRGeneratorForStatements::handleVariableReference(
 	Expression const& _referencingExpression
 )
 {
-	// TODO for the constant case, we have to be careful:
-	// If the value is visited twice, `defineExpression` is called twice on
-	// the same expression.
```

Comment:
I'm not quite sure I understand what this old comment warned about...

Reply:
It warned that you cannot just generate code for the expression like we did in the old codegen. Instead, I created a function that is called here and that contains the codegen for the expression only once.